components/threading/global_threads/thread_menu/thread_menu.tsx

```diff
@@ -59,6 +59,20 @@ function ThreadMenu({
 
     const isSaved = useSelector((state: GlobalState) => get(state, Preferences.CATEGORY_FLAGGED_POST, threadId, null) != null);
 
+    const handleReadUnread = useCallback(() => {
+        const lastViewedAt = hasUnreads ? Date.now() : unreadTimestamp;
+
+        dispatch(manuallyMarkThreadAsUnread(threadId, lastViewedAt));
+        dispatch(updateThreadRead(currentUserId, currentTeamId, threadId, lastViewedAt));
+    }, [
+        currentUserId,
```

Comment:
I don't know if it'll have as much impact on this component, but I've been looking a bit at where we pass props into a component purely for calling actions because that leads to the component potentially re-rendering more frequently for data that isn't actually needed for rendering. I'm not sure if this is the case here, but it might be something to keep in mind.

For example, assuming we weren't using `currentUserId` or some of the other props for rendering, we could refactor this into its own action to look more like
```javascript
function markThreadAsUnread(threadId) {
    return (dispatch, getState) => {
        const state = getState();
        const currentUserId = getCurrentUserId(state);
        const currentTeamId = getCurrentTeamId(state);
        const lastViewedAt = /*...*/;
        
        dispatch(manuallyMarkThreadAsUnread(threadId, lastViewedAt));
        dispatch(updateThreadRead(currentUserId, currentTeamId, threadId, lastViewedAt));
    };
}

function ThreadMenu({threadId}) {
    // ...
}
```
to reduce the props passed into ThreadMenu and keep it down to the ones that are only used for rendering